cn-create: add cn_cbor_float_create

vincent-d opened the pull request:
This adds ability to add a float:  `cn_cbor_float_create()`

Commit:
cn-create: add cn_cbor_double_create

cabo commented:
Thanks!

Shouldn't this be conditional on CBOR_NO_FLOAT?

I'm also wondering if people will be confused by calling 64-bit floating point values ("double" in C) "float".

vincent-d commented:
> Shouldn't this be conditional on CBOR_NO_FLOAT?

I guess so! Will fix.

> I'm also wondering if people will be confused by calling 64-bit floating point values ("double" in C) "float".

Maybe, indeed. Would you prefer two API calls (e.g. `cn_cbor_float_create()` and  `cn_cbor_double_create()`) or just renaming to `cn_cbor_double_create()`?

Commit:
test: add create double test

vincent-d commented:
- Added `cn_cbor_double_create()`. Let me know if you'd like to have a `cn_cbor_float_create()` as well.
- Changed to return `CN_CBOR_ERR_FLOAT_NOT_SUPPORTED` if built with `CBOR_NO_FLOAT`
- Added a test

cabo commented:
I think I'd go for not defining the API at all if CBOR_NO_FLOAT (which then leads to a compilation error).

Adding a `cn_cbor_float_create()` would make more sense if `struct cn_cbor` had a float member in the union; as it is, the conversion to double will just happen inside the function as opposed to the call site.  So I would add those at the same time if needed.

Commit:
cn-create: add cn_cbor_float_create

vincent-d commented:
- disable completely if `CBOR_NO_FLOAT`

- my last commit adds a float field in `struct cn_cbor`, but I wonder it it makes sense to keep it as the decoder does not use it...

cabo commented:
Right, this is not the complete story of adding binary32.  To do this efficiently, we'd need to split up `_write_double`.  But I think this is a stable intermediate state, so I'll merge this for now (on my way to a two-week vacation...).
